src/compiler/core/src/commonMain/kotlin/community/flock/wirespec/compiler/core/emit/ScalaEmitter.kt

```diff
@@ -78,10 +78,16 @@ open class ScalaEmitter(
         is Reference.Any -> "Any"
         is Reference.Custom -> value
         is Reference.Primitive -> when (type) {
-            Reference.Primitive.Type.String -> "String"
-            Reference.Primitive.Type.Integer -> "Long"
-            Reference.Primitive.Type.Number -> "Double"
-            Reference.Primitive.Type.Boolean -> "Boolean"
+            is Reference.Primitive.Type.String -> "String"
+            is Reference.Primitive.Type.Integer -> when(type.precision){
+                Reference.Primitive.Type.Precision._32 -> "Int"
```

Comment:
I think P32 and P64 are preferred instead of _32 and _64. Then the IDE doesn't suggest conforming to an Uppercase anymore :)